packages/astro/src/core/app/node.ts

```diff
@@ -23,6 +23,8 @@ function createRequestFromNodeRequest(req: IncomingMessage, body?: Uint8Array):
 		headers: new Headers(entries),
 		body: ['HEAD', 'GET'].includes(method) ? null : body,
 	});
+	// @ts-ignore
+	request.req = req;
```

Comment:
why is this needed?

Reply:
Oh, I guess for this reason:

> Also this change includes adding the original IncomingMessage (req) as a property of the Request that is created in createRequestFromNodeRequest, so that in Astro end-points there is a way to access the Express Session middleware.

We don't want to do that in the official adapter. You are of course welcome to do this in your own adapter. You could even build one that simply wraps the Node adapter.

We do want to find a solution for middleware, see https://github.com/withastro/roadmap/discussions/174

Reply:
I have done a local wrapper around the Node adapter to allow the Astro end-points to have more information from the Request provided that is in the original request. Just thought there where others in the community that might have the same need. Sorry, I will remove it and wait for the solution mentioned.

Reply:
No problem! Using a wrapper is the way to go, for now.